ocamltest/run_stubs.c

```diff
@@ -109,3 +114,56 @@ CAMLprim value caml_run_command(value caml_settings)
   caml_stat_free(settings.stderr_filename);
   CAMLreturn(Val_int(res));
 }
+
+CAMLprim value caml_drop_privilege(value name)
+{
+#ifdef __CYGWIN__
+  LUID privilege_luid;
+  DWORD dwReturnLength;
+
+  if (!LookupPrivilegeValue(NULL, String_val(name), &privilege_luid))
+    caml_raise_not_found();
+
+  /* Search through the primary token and, if this privilege is found, remove
+     it. */
+  if (!GetTokenInformation(GetCurrentProcessToken(), TokenPrivileges, NULL, 0,
+                           &dwReturnLength)
+      && GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
+    TOKEN_PRIVILEGES* process_privileges;
+
+    if (!(process_privileges = (TOKEN_PRIVILEGES*)malloc(dwReturnLength)))
+      caml_raise_out_of_memory();
+
+    if (GetTokenInformation(GetCurrentProcessToken(),
+                            TokenPrivileges, process_privileges,
+                            dwReturnLength, &dwReturnLength)) {
+      int i = 0;
+      LUID_AND_ATTRIBUTES* privilege = process_privileges->Privileges;
+
+      while (i++ < process_privileges->PrivilegeCount) {
```

Comment:
```suggestion
      LUID_AND_ATTRIBUTES* privilege = process_privileges->Privileges;

      for (int i = 0; i < process_privileges->PrivilegeCount; i++) {
```

Is there a reason you didn't opt for the `for` loop?

Reply:
Hah - because it was adapted from this in #8797:
```c
        /* Search for SeRestorePrivilege */
        while (i < privs->PrivilegeCount
               && (privilege->Luid.HighPart != privilege_luid.HighPart
                   || privilege->Luid.LowPart != privilege_luid.LowPart)) {
          i++;
          privilege++;
        }
```

(I can't resist the temptation to say ["Luxury! When I were a lad we didn't have any of these 'ere `for` loops!"](https://en.wikipedia.org/wiki/Four_Yorkshiremen))